src/test/java/org/prebid/server/bidder/medianet/MedianetBidderTest.java

```diff
@@ -38,8 +46,121 @@ public void setup() {
 
     @Test
     public void creationShouldFailOnInvalidEndpointUrl() {
-        assertThatIllegalArgumentException()
-                .isThrownBy(() -> new MedianetBidder("invalid_url", jacksonMapper));
+        assertThatIllegalArgumentException().isThrownBy(() -> new MedianetBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnErrorIfImpressionListSizeIsZero() {
+        //given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(emptyList())
+                .build();
+
+        //when
+        final Result<List<HttpRequest<BidRequest>>> result = medianetBidder.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors())
+                .allMatch(error -> error.getType() == bad_input
+                        && error.getMessage().equals("No valid impressions in the bid request"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerBidIfBannerIsPresentInRequestImp() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").banner(Banner.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), banner, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnVideoBidIfVideoIsPresentInRequestImp() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").video(Video.builder().build()).build()))
+                        .build(),
```

Comment:
looks like construction of sample BidRequest can be extracted to a separate function. Smth like `givenBidRequest(UnaryOperator<BidRequest> bidRequestBuilder)`.

then it can be called like `givenBidRequest(requestBuilder -> requestBuilder.vider(Video.builder().build()));`